Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabProcessorContext.cpp

```diff
@@ -30,27 +32,39 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
         : m_sourceUuid(sourceUuid)
     {}
 
-    bool PrefabProcessorContext::AddPrefab(AZStd::string prefabName, PrefabDom prefab)
+    bool PrefabProcessorContext::AddPrefab(PrefabDocument&& document)
     {
-        auto result = m_prefabs.emplace(AZStd::move(prefabName), AZStd::move(prefab));
-        return result.second;
+        AZStd::string name = document.GetName();
+        if (!m_prefabNames.contains(name))
+        {
+            m_prefabNames.emplace(AZStd::move(name));
+            PrefabContainer& container = m_isIterating ? m_pendingPrefabAdditions : m_prefabs;
```

Comment:
Can you add a quick comment to m_isIterating, AddPrefab and ListPrefab in PrefabProcessorContext.h as to what they are used for? I think it will make the code better understandable.

Reply:
Will do.